### ISSUE TITLE
Doesn't build with latest versions of dependencies

### ISSUE THREAD

**Issue by robbles** [harlow/kinesis-connectors#23]
I get the following errors when building with go 1.5.1 after running `go get github.com/harlow/kinesis-connectors`:

```
# kinesis_worker/vendor/github.com/harlow/kinesis-connectors
vendor/github.com/harlow/kinesis-connectors/logger.go:16: undefined: "kinesis_worker/vendor/github.com/go-kit/kit/log".NewPrefixLogger
vendor/github.com/harlow/kinesis-connectors/pipeline.go:31: cannot use "us-east-1" (type string) as type *string in field value
vendor/github.com/harlow/kinesis-connectors/pipeline.go:34: unknown "kinesis_worker/vendor/github.com/aws/aws-sdk-go/service/kinesis".GetShardIteratorInput field 'ShardID' in struct literal
vendor/github.com/harlow/kinesis-connectors/s3_emitter.go:28: cannot use "us-east-1" (type string) as type *string in field value
```

I think it likely does build against older versions of the deps. If you can post the versions you currently build it against, I'd be able to contribute a `gvt` manifest to install the correct ones. Or any of the other dependency management tool formats, not really set on any particular one.

**Comment by harlow** [comment]
thanks @robbles will take a look at this tonight

**Comment by robbles** [comment]
@harlow thanks! I'm definitely interested in using this project, so let me know how I can help.

I'm also curious about how the ezoic/kinesis-connectors fork relates to this project. Are you working together? It looks like that fork fixes these issues and adds some more features, but this version has been worked in parallel on in the meantime...?

**Comment by harlow** [comment]
@robbles great question on the fork. I haven't been using Kinesis in a while so there is a good chance that ezoic's fork has some good improvements. 

We worked to merge a bunch of his changes in a few months back:
https://github.com/harlow/kinesis-connectors/commits?author=superbaddude

It looks as though he's been putting a bunch more work in since then. It might be worth checking out his version and seeing if it works for you. I'll take a look and see if there is a clean path to merge his changes back into mainline.

**Comment by harlow** [comment]
@robbles I simplified the interface a little to feel more like the http handler package. Let me know if you need any help getting up and running. What's your use case once the data is off the stream?

The handler func takes in a `connector.Buffer` which has a `GetRecords` function which will have the records from Kinesis. From there they can be pushed to S3, aggregated, whatever needs to be done.

```go
func main() {
  var(
    appName = flag.String("a", "", "The app name")
    streamName = flag.String("s", "", "The stream name")
  )
  flag.Parse()

  c := connector.NewConsumer(*appName, *streamName)
  c.Start(connector.HandlerFunc(func(b connector.Buffer) {
    // fmt.Println(b.GetRecords())
    // process the records
  }))

  select {}
}
```

**Comment by robbles** [comment]
@harlow thanks, I'll give it a shot. At first glance this looks much easier to use!

The main use case is aggregating individual event records in Redshift for large analytics queries, possibly Elasticsearch as well at some point.

Have you used this with multiple distinct emitters (that don't simply depend on each other like Redshift/S3)? Would it make more sense to have some kind of MultiEmitter wrapper, or just run as separate apps?

**Comment by harlow** [comment]
Might be worth checking out the Kinesis Firehose project for S3/Redshift
integration. Looks pretty stable and reasonably priced

**Comment by robbles** [comment]
I have - it looks like a great option, but sadly it's not available in the region we're running in.

**Comment by harlow** [comment]
I'm going to close this issue for the time being as the project should build correctly now. Feel free to re-open or start another issue if you run into anything else.

**Comment by robbles** [comment]
@harlow I would still recommend having some sort of vendoring or pinning of package versions, otherwise the build is going to break again when the APIs of the third-party packages change. 

I've been using https://github.com/FiloSottile/gvt as a minimally disruptive vendoring tool for this. It's possible to just commit a manifest of the current versions of packages that are in use, and ignore it when building in your local setup. That way a snapshot of the working package versions is stored somewhere.

I can make a PR with only the manifest if you're interested.

**Comment by harlow** [comment]
Sounds great. Would def appreciate a PR for that.
